aries_cloudagent/messaging/models/base_record.py

```diff
@@ -374,34 +371,20 @@ async def delete_record(self, session: ProfileSession):
             await storage.delete_record(self.storage_record)
         # FIXME - update state and send webhook?
 
-    @property
-    def webhook_payload(self):
-        """Return a JSON-serialized version of the record for the webhook."""
-        return self.serialize()
-
-    @property
-    def webhook_topic(self):
-        """Return the webhook topic value."""
-        return self.WEBHOOK_TOPIC
-
-    async def send_webhook(
-        self, session: ProfileSession, payload: Any, topic: str = None
-    ):
-        """Send a standard webhook.
+    async def emit_event(self, session: ProfileSession, payload: Any):
+        """Emit an event.
 
         Args:
             session: The profile session to use
-            payload: The webhook payload
-            topic: The webhook topic, defaulting to WEBHOOK_TOPIC
+            payload: The event payload
         """
-        if not payload:
+
+        if not self.RECORD_TOPIC or not self.state or not payload:
```

Comment:
Calling out the logic I'm seeing to validate intent: so if the BaseRecord subclass does not define a `RECORD_TOPIC` or the state has not been correctly set on the record, or if no payload is given, no event will be emitted. Looking at the diff, this is what was happening before as well, just simplified.

Reply:
Yes, same intent as before, we don't want to send events if there is not topic+state+payload. Before, there was no state and you could pass in a webhook topic to override.